### ISSUE TITLE
Using code=_NULL in dataset search?

### ISSUE THREAD

**Issue by gdower** [CatalogueOfLife/backend#1275]
Does _NULL only work for elastic search like in your example? 

http://api.checklistbank.org/dataset/3/nameusage/search?nomCode=_NULL&facet=rank&facet=SECTOR_DATASET_KEY

It would be useful if we could do something like:

http://api.checklistbank.org/dataset?code=_NULL&contributesTo=3

**Comment by gdower** [comment]
Actually just realized that the dataset code parameter doesn't work right anyway with only 2 datasets returning as having zoological set when we have much more than that, but had previously opened a ticket on it:

http://api.checklistbank.org/dataset?code=zoological&contributesTo=3
http://api.checklistbank.org/dataset/1027/settings <- not in the list above

https://github.com/CatalogueOfLife/backend/issues/1200

**Comment by mdoering** [comment]
Yes that only works in ES searches. Agree it would be nice to use it elsewhere, but from all the work to be done this likely falls in lower priorities for now

**Comment by mdoering** [comment]
It is difficult to add that feature cause the frameworks interpret the code value to be one of the enumeration values. `_NULL` is throwing bad argument exceptions very early on. But I have added a new query param `codeIsNull=true` that does the same and was easy to implement. It is not nice to treat this special, so I have deprecated its use and I want to find a consistent long term solution for this and all other parameters. But for now we can at least search for the missing datasets

**Comment by gdower** [comment]
Thanks!
